### PR TITLE
[Review Only] Add InVMAccessControlProfile for GalleryRP in new API version 2024-03-03

### PR DESCRIPTION
**Pull request by lkumaar**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by lkumaar** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by lkumaar** [comment]
Requesting review for new resource type to unblock manifest check in.
1af11c441e66d5b3ae05f801325d2d6d27749759 has the main changes, rest are housekeeping for new API version

File: specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/examples/galleryResourceProfileExamples/GalleryInVMAccessControlProfileVersion_Delete.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    "galleryName": "myGalleryName",
+    "inVMAccessControlProfileName": "myInVMAccessControlProfileName",
+    "inVMAccessControlProfileVersionName": "1.0.0"
+  },
+  "responses": {
+    "200": {
+      "headers": {
+        "Azure-AsyncOperation": "https://management.azure.com/subscriptions/000000-8de3-42e7-b00e-8970003527749a/providers/Microsoft.Compute/locations/westus/capsOperations/0012b61c-2d36-40bc-b7ed-1f0e48757277?api-version=2024-03-03",
+        "Location": "https://management.azure.com/subscriptions/000000-8de3-42e7-b00e-8970003527749a/providers/Microsoft.Compute/locations/westus/capsOperations/0012b61c-2d36-40bc-b7ed-1f0e48757277?monitor=true&api-version=2024-03-03"
+      }
```
Comment (raosuhas) [inline specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/examples/galleryResourceProfileExamples/GalleryInVMAccessControlProfileVersion_Delete.json:15]: 200 \204 should be sync .. not location headers\ async operations

File: specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/examples/galleryResourceProfileExamples/GalleryInVMAccessControlProfile_Delete.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "202": {
+      "headers": {
+        "Azure-AsyncOperation": "https://management.azure.com/subscriptions/000000-8de3-42e7-b00e-8970003527749a/providers/Microsoft.Compute/locations/westus/capsOperations/0012b61c-2d36-40bc-b7ed-1f0e48757277?api-version=2024-03-03",
+        "Location": "https://management.azure.com/subscriptions/000000-8de3-42e7-b00e-8970003527749a/providers/Microsoft.Compute/locations/westus/capsOperations/0012b61c-2d36-40bc-b7ed-1f0e48757277?monitor=true&api-version=2024-03-03"
+      }
+    },
+    "204": {
+      "headers": {
+        "Azure-AsyncOperation": "https://management.azure.com/subscriptions/000000-8de3-42e7-b00e-8970003527749a/providers/Microsoft.Compute/locations/westus/capsOperations/0012b61c-2d36-40bc-b7ed-1f0e48757277?api-version=2024-03-03",
+        "Location": "https://management.azure.com/subscriptions/000000-8de3-42e7-b00e-8970003527749a/providers/Microsoft.Compute/locations/westus/capsOperations/0012b61c-2d36-40bc-b7ed-1f0e48757277?monitor=true&api-version=2024-03-03"
```
Comment (raosuhas) [inline specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/examples/galleryResourceProfileExamples/GalleryInVMAccessControlProfile_Delete.json:25]: same comment here

File: specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/examples/galleryResourceProfileExamples/GalleryInVMAccessControlProfileVersion_Delete.json (hunk 1 above)
Comment (lkumaar) [reply specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/examples/galleryResourceProfileExamples/GalleryInVMAccessControlProfileVersion_Delete.json]: yeah... fixed. will fix other examples before the actual swagger PR

File: specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/examples/galleryResourceProfileExamples/GalleryInVMAccessControlProfile_Delete.json (hunk 2 above)
Comment (lkumaar) [reply specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/examples/galleryResourceProfileExamples/GalleryInVMAccessControlProfile_Delete.json]: fixed

File: specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json [hunk 3]
```diff
@@ -1737,6 +1737,518 @@
...
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "galleryName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Shared Image Gallery in which the InVMAccessControlProfile it to be deleted."
```
Comment (RajalakshmiSwaminathan) [inline specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json:1945]: The name of the Shared Image Gallery in which the InVMAccessControlProfile resides.

Comment (lkumaar) [reply specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json]: updated

File: specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json [hunk 4]
```diff
@@ -0,0 +1,4101 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/inVMAccessControlProfiles/{inVMAccessControlProfileName}": {
+      "put": {
+        "tags": [
+          "GalleryInVMAccessControlProfiles"
+        ],
+        "operationId": "GalleryInVMAccessControlProfiles_CreateOrUpdate",
```
Comment (Sandido) [inline specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json:1746]: This always confused me. Why is there a CreateOrUpdate operation, then a separate Update operation? Does the CreateOrUpdate actually do an Update? Do these operations do something different or under different circumstances?

File: specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json [hunk 5]
```diff
@@ -0,0 +1,4101 @@
...
+          "type": "string",
+          "description": "This property allows you to specify whether the access control rules are in Audit mode, in Enforce mode or Disabled. Possible values are: 'Audit', 'Enforce' or 'Disabled'.",
+          "enum": [
+            "Audit",
+            "Enforce",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "name": "AccessControlRulesMode",
+            "modelAsString": false
```
Comment (Sandido) [inline specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json:3875]: This must be set to true, for all enums here.

File: specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json [hunk 6]
```diff
@@ -0,0 +1,4101 @@
...
+        "defaultAccess": {
+          "type": "string",
+          "description": "This property allows you to specify if the requests will be allowed to access the host endpoints. Possible values are: 'Allow', 'Deny'.",
+          "enum": [
+            "Allow",
+            "Deny"
+          ],
+          "x-ms-enum": {
+            "name": "EndpointAccess",
+            "modelAsString": false
```
Comment (Sandido) [inline specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json:3887]: set to true

**Review by Sandido — CHANGES_REQUESTED** [review]

File: specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json (hunk 5 above)
Comment (lkumaar) [reply specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json]: set to true

File: specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json (hunk 6 above)
Comment (lkumaar) [reply specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json]: set to true

File: specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json (hunk 4 above)
Comment (lkumaar) [reply specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json:1746]: > Does the CreateOrUpdate actually do an Update?

Yes

> Do these operations do something different or under different circumstances?

No. Both operations are handled in the exact same manner in the service side. Both operations follow patch semantics, and this is the usual pattern for all our resources. I believe it's the same with other RPs in Compute.

**Pull request closed without merging**